sql/parser/builtins.go

```diff
@@ -83,16 +84,16 @@ func (b builtin) match(types typeList) bool {
 // functions.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
+	"length": {stringBuiltin(1, func(s []string) (Datum, error) {
```

Comment:
Looks like the string builtins either have 1, 2 or 3 args. Might be better to have separate `stringBuiltin{1,2,3}` helpers so that the implementation function gets precisely the number of arguments it wants.